src/core/qfieldappauthrequesthandler.cpp

```diff
@@ -0,0 +1,177 @@
+{
+
+}
+
+void QFieldAppAuthRequestHandler::enterCredentials( const QString realm, const QString username, const QString password )
+{
+  QgsCredentials::instance()->put( realm, username, password );
+}
+
+QString QFieldAppAuthRequestHandler::getFirstUnhandledRealm() const
+{
+  for ( const RealmEntry &entry : qgis::as_const( mRealms ) )
+  {
+    if ( !entry.canceled )
+    {
+      return entry.realm;
+    }
+  }
+  return QString();
+}
+
+bool QFieldAppAuthRequestHandler::handleLayerLogins()
+{
+  if ( !getFirstUnhandledRealm().isEmpty() )
+  {
+    emit showLoginDialog( getFirstUnhandledRealm() );
+
+    connect( this, &QFieldAppAuthRequestHandler::loginDialogClosed, [ = ]( const QString &realm, bool canceled )
+    {
+      if ( canceled )
+      {
+        //realm not successful handled - but canceled
+        for ( int i = 0; i < mRealms.count(); i++ )
+        {
+          if ( mRealms.at( i ).realm == realm )
+          {
+            mRealms.replace( i, RealmEntry( realm, true ) );
+            break;
+          }
+        }
+      }
+      else
+      {
+        //realm successful handled (credentials saved) - remove realm
+        for ( int i = 0; i < mRealms.count(); i++ )
+        {
+          if ( mRealms.at( i ).realm == realm )
+          {
+            mRealms.removeAt( i );
+            break;
+          }
+        }
+      }
+
+      if ( !getFirstUnhandledRealm().isEmpty() )
+      {
+        //show dialog as long as there are unhandled realms
+        emit showLoginDialog( getFirstUnhandledRealm() );
+      }
+      else
+      {
+        emit reloadEverything();
+      }
+    } );
+  }
+  else
+  {
+    return false;
+  }
+  return true;
+}
+
+void QFieldAppAuthRequestHandler::clearStoredRealms()
+{
+  mRealms.clear();
+}
+
+void QFieldAppAuthRequestHandler::authNeeded( const QString realm )
+{
+  for ( const RealmEntry &entry : qgis::as_const( mRealms ) )
+  {
+    if ( entry.realm == realm )
+    {
+      //realm already in list
+      return;
+    }
+  }
+
+  RealmEntry unhandledRealm( realm );
+  mRealms << unhandledRealm;
+}
+
+void QFieldAppAuthRequestHandler::handleAuthRequest( QNetworkReply *reply, QAuthenticator *auth )
+{
+  Q_ASSERT( qApp->thread() == QThread::currentThread() );
+
+  QString username = auth->user();
+  QString password = auth->password();
+
+  if ( username.isEmpty() && password.isEmpty() && reply->request().hasRawHeader( "Authorization" ) )
+  {
+    QByteArray header( reply->request().rawHeader( "Authorization" ) );
+    if ( header.startsWith( "Basic " ) )
+    {
+      QByteArray auth( QByteArray::fromBase64( header.mid( 6 ) ) );
+      int pos = auth.indexOf( ':' );
+      if ( pos >= 0 )
+      {
+        username = auth.left( pos );
+        password = auth.mid( pos + 1 );
+      }
+    }
+  }
+
+  for ( ;; )
+  {
+    bool ok = QgsCredentials::instance()->get(
+                QStringLiteral( "%1 at %2" ).arg( auth->realm(), reply->url().host() ),
```

Comment:
oops - i just merged it

Reply:
It's the realm at the host. In german it would work as well, but it's english. Maybe "auf". So the response is, yes it might should be translated. But not that bad if not.

Reply:
I can do it in another PR.

Reply:
Oh no, I've been confused. This is not something that should be translated. It's the combination stored in the Credentials. This is done like that in QGIS as well. 

Nothing has to be done...